doc/source/tutorials/compiler.rst

```diff
@@ -25,46 +32,16 @@ Creating an instance of the backend provides access to these objects:
 
     <class 'qibolab.compilers.compiler.Compiler'>
 
-The transpiler is responsible for transforming any circuit to one that respects
-the chip connectivity and native gates. The compiler then transforms this circuit
-to the equivalent pulse sequence. Note that there is no transpiler in Qibolab, therefore
-the backend can only execute circuits that contain native gates by default.
-The user can modify the compilation process by changing the  ``compiler`` attributes of
-the ``QibolabBackend``.
 
-In this example, we executed circuits using the backend ``backend.execute_circuit`` method,
-unlike the previous example (:ref:`tutorials_circuits`) where circuits were executed directly using ``circuit(nshots=1000)``.
-It is possible to perform compiler manipulation in both approaches.
-When using ``circuit(nshots=1000)``, Qibo is automatically initializing a ``GlobalBackend()`` singleton that is used to execute the circuit.
-Therefore the previous manipulations can be done as follows:
+The default compiler provides rules for the following single-qubit gates:
+``Z``, ``RZ``, ``I``, ``GPI``, ``GPI2``, ``M``
+and the following two-qubit gates: ``CZ``, ``CNOT`` subject to platform availability.
+The user can modify the compilation process by defining custom compilation rules and registering
+them to the  ``compiler`` attribute of the ``QibolabBackend``.
+The default rules defined in :py:mod:`qibolab.compilers.default` can serve as an example.
 
-.. testcode:: python
-
-    import qibo
-    from qibo import gates
-    from qibo.models import Circuit
-    from qibo.backends import GlobalBackend
-
-    # define circuit
-    circuit = Circuit(1)
-    circuit.add(gates.GPI2(0, 0.1))
-    circuit.add(gates.M(0))
-
-    # set backend to qibolab
-    qibo.set_backend("qibolab", platform="dummy")
-
-    # execute circuit
-    result = circuit(nshots=1000)
-
-
-Defining custom compiler rules
-==============================
-
-The compiler can be modified by adding new compilation rules or changing existing ones.
-As explained in :ref:`main_doc_compiler` section, a rule is a function that accepts a Qibo gate and a Qibolab platform
-and returns the pulse sequence implementing this gate.
-
-The following example shows how to modify the compiler in order to execute a circuit containing a Pauli X gate using a single pi-pulse:
+The following example shows how to modify the compiler in order to execute a circuit containing an ``X`` gate
+directly using a single pi-pulse, instead of converting it to ``GPI2``:
```

Comment:
I was also not sure whether I should update or drop this. I updated since it didn't require much effort and in the end kept it, since, even though we agreed to not make this part of the "public API" I believe that what is written here actually works. So unless we are expecting to break it in the future (in some 0.2.*), it wouldn't hurt and shouldn't require maintenance either. I am also okay with dropping though.

Reply:
~~If it was not for the update, I would have a clear reply: drop it, and then we will resurrect as soon as the `Compiler` will be exposed.~~

At a second thought: we could always search for it with Git, so I'd say drop it, and whenever we will re-release `Compiler` as part of the public API we will search for the updated file (I have done it seldom, but I should be able to find it within Git's history - i.e. without scanning manually commit by commit).

Reply:
File removed in 8e23cb994ebb11a4cf0d60002548393cc7e4ea7a. I did the removal alone in a commit, so I believe, if needed, we should be able to recover the file just by reverting this commit.